src/services/NotificationService.ts

```diff
@@ -0,0 +1,55 @@
+import axios from 'axios';
+import { UserSubscription } from '../core/users/User';
+
+type RequestData = {
+  event: string;
+  payload: Record<string, any>;
+  clientId: string;
+  email?: string;
+};
+
+export class Notifications {
+  private static instance: Notifications;
+
+  static getInstance(): Notifications {
+    if (Notifications.instance) {
+      return Notifications.instance;
+    }
+
+    Notifications.instance = new Notifications();
+
+    return Notifications.instance;
+  }
+
+  subscriptionChanged({
+    clientId,
+    subscription,
+    space,
+  }: {
+    clientId: string;
+    subscription: UserSubscription;
+    space: number;
+  }): Promise<void> {
+    return this.post({ event: 'SUBSCRIPTION_CHANGED', payload: { subscription, space }, clientId });
+  }
+
+  private async post(data: RequestData): Promise<void> {
+    try {
+      const res = await axios.post(process.env.NOTIFICATIONS_URL as string, data, {
+        headers: { 'X-API-KEY': process.env.NOTIFICATIONS_API_KEY as string },
+      });
+      if (res.status !== 201)
+        console.warn(
+          `Post to notifications service failed with status ${res.status}. Data: ${JSON.stringify(data, null, 2)}`,
+        );
+    } catch (err) {
+      console.warn(
```

Comment:
couldn't find a better way to use fastify.log here. Perhaps we could export the Fastify instance as a singleton from `app.ts`

Defaulted to console.

Reply:
we will revisit this in the future